Source/objects.cpp

```diff
@@ -818,26 +818,15 @@ void __fastcall AddL2Objs(int x1, int y1, int x2, int y2)
 
 void __fastcall AddL3Objs(int x1, int y1, int x2, int y2)
 {
-	int v4;  // edi
-	int *v5; // esi
-	int v6;  // ebx
-	int x;   // [esp+0h] [ebp-8h]
-	int y;   // [esp+4h] [ebp-4h]
-
-	x = x1;
-	for (y = y1; y < y2; ++y) {
-		v4 = x;
-		if (x < x2) {
-			v5 = (int *)((char *)dPiece + 4 * (y + 112 * x));
-			do {
-				v6 = *v5;
-				if (*v5 == 531)
-					AddObject(OBJ_L3LDOOR, v4, y);
-				if (v6 == 534)
-					AddObject(OBJ_L3RDOOR, v4, y);
-				++v4;
-				v5 += 112;
-			} while (v4 < x2);
+	int i, j, pn;
+
+	for (i = y1; i < y2; i++) {
+		for (j = x1; j < x2; j++) {
+			pn = dPiece[j][i];
+			if (pn == 531)
+				AddObject(OBJ_L3LDOOR, j, i);

```

Comment:
I know I've already merged #748 and #749. However, there is one aspects of these three PRs that I think we should discuss.

Calling `AddObject` with `j, i` instead of `i, j` *looks wrong*, if you know what I mean. Since `y` follows `x` in the alphabet, I think the outer loop should use `j` and the inner loop use `i`.

Also, I think this is a pattern we should apply to other parts of the code that has been cleaned up. Of course, I can be persuaded otherwise, given the right rationale. :)

Reply:
:P

Reply:
> :P

@qndel your argument is quite persuasive ;)